### PR TITLE
Fix provider configuration code to handle Unknown values correctly

### PR DESCRIPTION
**Pull request by SarahFrench**
Fixes https://github.com/hashicorp/terraform-provider-google/issues/14444

Related to https://github.com/GoogleCloudPlatform/magic-modules/pull/8798

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
provider: addressed a bug where configuring the provider with unknown values did not behave as expected
```

### CODE REVIEW

**Comment by SarahFrench** [comment]
I've yet to manually test this change to see if this use case from the original issue works as expected again:

```
provider "google-beta" {
  project     = "foo"
  credentials = base64decode(google_service_account_key.terraform_service_account.private_key)
}
```

File: mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb [hunk 1]
```diff
@@ -120,14 +120,16 @@ func TestFrameworkProvider_LoadAndValidateFramework_project(t *testing.T) {
...
 		// Handling unknown values
-		// TODO(SarahFrench) make these tests pass to address: https://github.com/hashicorp/terraform-provider-google/issues/14444
-		// "when project is an unknown value, the provider treats it as if it's unset (align to SDK behaviour)": {
-		// 	ConfigValues: fwmodels.ProviderModel{
-		// 		Project: types.StringUnknown(),
-		// 	},
-		// 	ExpectedDataModelValue:    types.StringNull(),
-		// 	ExpectedConfigStructValue: types.StringNull(),
-		// },
+		"when project is an unknown value, the provider treats it as if it's unset and uses an environment variable instead": {
```
Comment (rileykarson) [inline mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb:123]: The actual use case here is that eventually the value will be known and the provider will authenticate with the now-known value, right? Or did this just never work?

**Review by rileykarson — APPROVED** [review]

File: mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb (hunk 1 above)
Comment (SarahFrench) [reply mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb:123]: > eventually the value will be known and the provider will authenticate with the now-known value

Yeah this is right - in the original GH issue around the `credentials` field the SDK code would see the unknown value for credentials as an empty string and treated like it was not provided at all. During the process of applying a plan the provider re-authenticates multiple times, and eventually the value of credentials changes from `""`  to the credentials information. It's like 'just-in-time' credentials.

I imagine it's the same for `project`, but I don't think it's good practice to rely on this behaviour to be honest!

Comment (rileykarson) [reply mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb:123]: That tracks with me! And we're removing the surprising behaviour for both SDK + Framework with `5.0.0`, right?

Comment (SarahFrench) [reply mmv1/third_party/terraform/fwtransport/framework_config_test.go.erb:123]: We're removing how empty strings get ignored in 5.0.0 (e.g if a user sets project = "" in their provider block it'll stop the GOOGLE_PROJECT env being used in 5.0+), but there isn't a similar correction we need to make in 5.0 for unknown values, I think.

**Pull request merged**